.gitsplit.yml

```diff
@@ -0,0 +1,18 @@
+splits:
+    - prefix: "src/bundle"
+      target: "https://${GH_TOKEN}@github.com/scheb/2fa-bundle.git"
+    - prefix: "src/email"
+      target: "https://${GH_TOKEN}@github.com/scheb/2fa-email.git"
+    - prefix: "src/google"
+      target: "https://${GH_TOKEN}@github.com/scheb/2fa-google-authenticator.git"
+    - prefix: "src/totp"
+      target: "https://${GH_TOKEN}@github.com/scheb/2fa-totp.git"
+    - prefix: "src/backup"
+      target: "https://${GH_TOKEN}@github.com/scheb/2fa-backup-code.git"
+    - prefix: "src/trusted"
+      target: "https://${GH_TOKEN}@github.com/scheb/2fa-trusted-devices.git"
+
+origins:
+    - ^master$
+    - ^v\d+\.\d+$
```

Comment:
Just to make sure that I understand that right: This is pushing all releases made on the main repository down to the split packages. When there is no change in a split it will still create a tag, but on the latest commit. So in the split packages there can be multiple release tags on the same commit.

Reply:
> Just to make sure that I understand that right: This is pushing all releases made on the main repository down to the split packages.


Yes. The regex limit the split to the `master` branch, `vx.y` refs (e.g. `v2.5` branch) and vx.y.z refs (e.g. `v3.1.5` tag).

The sub-repository won’t change if there is no change in the associated folder.

> So in the split packages there can be multiple release tags on the same commit.

The tags will always be set even if there is no change. This means that when you release the `v1.2.3` every sub-packages will have a tag `v1.2.3`. Please read [this issue](https://github.com/dependabot/feedback/issues/758) if you or your users have dependabot installed. This is not a real problem, but could be a bit annoying